scripts/dist-management/download-dist.js

```diff
@@ -0,0 +1,98 @@
+const path = require('path');
+const fs = require('fs');
+
+const execa = require('execa');
+const request = require('request');
+const unzipper = require('unzipper');
+
+const getCurrentCommitHash = async () => {
+    const { stdout: hash } = await execa('git', ['rev-parse', '--verify', 'HEAD']);
+
+    return hash;
+};
+
+const repoClean = async () => {
+    const { stdout } = await execa('git', ['status', '--short']);
+
+    if (stdout !== '') {
+        throw new Error('Repository is not clean');
```

Comment:
Should we allow this even if the repo is not clean? I could easily see someone making some edits before trying to build for the first time. Of course then we'd need some other way to verify we don't run this on subsequent builds...

Reply:
I debated doing this because we also have the `tsconfig.tsbuildinfo` files but decided against including it on this PR because we will have to do something a bit more clever such as:

* download the assets
* identify the packages that have been modified
* build those packages

We want to improve how we detect the packages that need to be built so I was planning on doing this (or similar) at the same time we take care of that.

Reply:
Fair enough, I'm okay tackling this separately - let's just make sure we track it. In the meantime even supporting this only on a clean build is a vast improvement compared the current state of the world.

Reply:
https://app.zenhub.com/workspaces/webhint-5bc4f59f4b5806bc2bed23d0/issues/webhintio/hint/2095#issuecomment-538697074